docs/developers.rst

```diff
@@ -0,0 +1,7 @@
+Developers guide
+================
+
+.. todo::
+
+    Everything developers need to know to get a dev environment running,
```

Comment:
@perrygeo for now, this could crosslink to some of the wiki pages: 

* https://github.com/mapbox/rasterio/wiki/Development-Guide
* https://github.com/mapbox/rasterio/wiki/Exposing-GDAL-Functionality
* https://github.com/mapbox/rasterio/wiki/Cython-and-GDAL

Though a longer term goal of consolidating that information into the docs is a good idea.